### PR TITLE
Add Store Widgets target and Fastlane configuration

### PR DESCRIPTION
**Pull request by mokagio**
### Description

Add a new widgets extension target and configures the Fastlane automation to code sign it. Internal ref pdnsEh-yV-p2.

The diff is unfortunately long, as a byproduct of adding the target. However, all the code in `StoreWidgets` was generated by Xcode and I didn't modify it's functionality. So, the only files to review in this PR are `Fastfile` and the `project.pbxproj`. Conveniently, they are both at the very bottom of the diff.

### Testing instructions

If CI succeeds in building the Enterprise build, then we're good.

I also created a hack PR to test that CI can build the App Store build [here](https://github.com/woocommerce/woocommerce-ios/pull/7624)—It passed.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. – _The addition of the widgets target is a major one, but this PR has not feature change from that point of view._

### CODE REVIEW

File: WooCommerce/StoreWidgets/StoreWidgets.swift
Comment (mokagio) [inline WooCommerce/StoreWidgets/StoreWidgets.swift]: Apple having inconsistent style _in the same template file!_

File: fastlane/Fastfile
Comment (mokagio) [inline fastlane/Fastfile:947]: I moved this around to keep it consistent with the one below. 

I moved `readonly: true` at the bottom below to have a parameter less likely to change at the end of the list and avoid awkward diffs where the only change is adding or removing a `,` to the parameter that used to be last but no longer is.

File: fastlane/Fastfile
Comment (mokagio) [inline fastlane/Fastfile]: I'm not sure whether more detail is needed here? I'm not keen in adding links to internal documents (even if in shorthand form) in the code. There should be detail here to search for the information internally if necessary. What do you think?

**Comment by mokagio** [comment]
Note that the [Installable Builds failure](https://buildkite.com/automattic/woocommerce-ios/builds/7537#0182ec81-951b-4c8b-977d-0c7845cefde2) is expected at this time because of unrelated issues accessing the Enterprise account.

image

File: fastlane/Fastfile
Comment (mokagio) [inline fastlane/Fastfile]: As expected, the build failed during code signing:

image

```
error: StoreWidgetsExtension has conflicting provisioning settings. StoreWidgetsExtension is automatically signed for development, 
but a conflicting code signing identity iPhone Distribution has been manually specified.
Set the code signing identity value to "Apple Development" in the build settings editor, or switch to manual signing in the Signing & Capabilities editor.
(in target 'StoreWidgetsExtension' from project 'WooCommerce')
```

File: fastlane/Fastfile
Comment (mokagio) [inline fastlane/Fastfile:42]: In WordPress iOS, we have a [fancier, DRYer setup](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/fastlane/Fastfile#L30-L43) for the bundle identifiers. I didn't port it across because I found it less readable than this one. It makes sense in WordPress where we have 9 extensions, but I think this works fine here.

Happy to update in a dedicated PR if you feel it would be good for future proofing.

File: fastlane/Fastfile
Comment (AliSoftware) [reply fastlane/Fastfile]:
```suggestion
      # For a time, we used to have this parameter specifying a entitlements.
      # In August 2022, `match` started being unable to generate new
      # provisioning profiles and the only solution I found was to remove it.
      #
      # Leaving it here for future reference as something to try if `match`
      # gives us trouble again.
      # template_name: 'Apple Pay Pass Suppression'
```

Comment (AliSoftware) [reply fastlane/Fastfile]: At first I misread (multiple times, even… ☕ ) this as "`match` started being **able** to […]" and that sentence got me so confused, because I was like "so wait, match got better with a new feature, and you had to find a way to… fix it?". Stumbled me for two good minutes… until I was about to make a GitHub suggestion to reword this and finally notice that you wrote "unable", not "able".
I think the "starting being" wording led me astray, as I'm not used in hearing "start to be unable" and would expect a wording like "stop being able" instead.

Also I'm not sure if that wording is supposed to mean that this is a bug in `match` (it started to have a bug which made it fail to create new profiles) or a feature (they purposely removed that ability because it wasn't working as expected or whatnot, and thus removed the `template_name` parameter as well).

Your subsequent sentence suggesting that we might re-introduce the `template_name` parameter again in the future lead me to finally understand that this might probably be a bug and not a conscious decision by fastlane, but the "to try if match gives us trouble again" added to the confusion 😅  As in: if the bug was when match tried to create new profiles because of this parameter, how come match would… give us trouble again without it? If anything, I feel that the reason why we'd want to add the parameter back is if match fixed the profile creation bug and we wanted to give it back a try, no?

**Review by Ecarrion — APPROVED** [review]
I can't comment much about the `fastfile` but the setup is working correctly for my initial widget UI. https://github.com/woocommerce/woocommerce-ios/pull/7584

File: fastlane/Fastfile
Comment (mokagio) [reply fastlane/Fastfile]: The truth is... I don't know if this is a bug in `match`, a bug on Apple's side, or something that never worked, was worked around in a different way, and we only found out about it now because we decided to generate new profiles 🙃 

It could be the latter, because the [automation](https://github.com/Automattic/match-regenerator/blob/edef2ca14b0f62ccfab0f6426711daf296d56b98/fastlane/Fastfile) I cobbled together in the past to regenerate the provisioning profiles across all app never had `template_name` and worked back then. I had forgotten about that code and only though of it in writing this.

I think I'm just going to remove the comment. Git remembers.

**Comment by mokagio** [comment]
@Ecarrion thank you for looking into this. I think if we merge it as is, the dummy widget will be available to users. That would be confusing to say the least. 

Might it be best to merge this only when the widget is usable enough?

File: fastlane/Fastfile
Comment (AliSoftware) [reply fastlane/Fastfile]: 👍 on removing the comment. git blame ftw 😛

**Comment by Ecarrion** [comment]
Hey @mokagio , I think we can merge this as is, as we already have an open PR that hides the widget under a local feature flag.

**Comment by Ecarrion** [comment]
@mokagio I'm going to merge this so we can advance with the development on trunk! Thanks a lot again! 🙇

**Comment by AliSoftware** [comment]
@mokagio seems that `trunk` broke after this PR landed… At least some people in Slack (@jaclync, @joshheald), are mentioning they can't build the app on device in trunk due to provisioning profile for the widget target is missing. (ref: p1662368046975199-slack-CC7L49W13). Any idea?

**Pull request merged**
